internal_coroutines_test.go

```diff
@@ -1034,3 +1034,53 @@ func TestDecodeFutureChain(t *testing.T) {
 	}
 	require.EqualValues(t, expected, history)
 }
+
+func TestSelectFuture_WithBatchSets(t *testing.T) {
+	var history []string
+	d := newDispatcher(background, func(ctx Context) {
+		future1, settable1 := NewFuture(ctx)
+		future2, settable2 := NewFuture(ctx)
+		future3, settable3 := NewFuture(ctx)
+
+		s := NewSelector(ctx)
```

Comment:
hmm...  in the example in the playground, i am writing twice on to channel and hence select executes twice. If i write only once then the select is blocked.
In case of cadence.Future, setting more than once would panic, so  it makes sense that we can only set once on future and that means we can only select once on the same future. 
Unblocking 3 times would add confusion since future is not allowed to set 3 times.

To be fair with our logic that we panic on second Set on future, we can as well panic on second Select on the same future instead of blocking. any thoughts?